dojo/tools/semgrep/parser.py

```diff
@@ -0,0 +1,51 @@
+import json
+from dojo.models import Finding
+from dojo.tools.semgrep.models import SemgrepJSONResult
+
+
+class SemgrepJSONParser(object):
+
+    def __init__(self, filehandle, test):
+        tree = self.parse_json(filehandle)
+
+        self.items = []
+        if tree:
+            results = tree.get('results')
+
+            if not results:
+                return
+
+            for item in results:
+                title = item['check_id']
+                f = SemgrepJSONResult(item['extra'], item['path'], item['start'], item['end'])
+
+                findingItem = Finding(
+                    title=f.title,
+                    severity=f.severity,
+                    numerical_severity=Finding.get_numerical_severity(f.severity),
+                    description=f.message,
```

Comment:
In my test file, I have a message that contains quite a few `\n`, which oddly breaks the description in defectdojo. Do you think it'd be a good thing to ensure newline characters are removed before being inserted into DefectDojo?

My example
```
Found object deserialization using ObjectInputStream. Deserializing entire\nJava objects is dangerous because malicious actors can create Java object\nstreams with unintended consequences. Ensure that the objects being deserialized\nare not user-controlled. If this must be done, consider using HMACs to sign\nthe data stream to make sure it is not tampered with, or consider only\ntransmitting object fields and populating a new object.\n"
```

Reply:
Really don"t know. I did not have the case in my whole tests .